File: test/pvo_file.c

/// vim: tabstop=4:expandtab:hlsearch

#include "../pvo.h"
#include "../pvo_report.h"
#include "../pvo_memory.h"

#include <iso646.h>

int main( int argc, char** argv ) {
    pvo_cookie_t cookie;
    pvo_file_t   fh;
    void* p = NULL;

    MPI_Init( &argc, &argv );
    pvo_init( MPI_COMM_WORLD );

    if( -1 == pvo_cookie_create( pvo_world_rank()%2, &cookie ))
        PVO_DIE( "pvo_cookie_create() failed." );

    if( -1 == pvo_cookie_insert_var( cookie, PVO_VAR_NODEDATA, PVO_VAR_INT32  , 8, "X", p ))
        PVO_DIE( "pvo_cookie_insert_var() failed." );
    if( -1 == pvo_cookie_insert_var( cookie, PVO_VAR_CELLDATA, PVO_VAR_FLOAT32, 1, "Y", p ))
        PVO_DIE( "pvo_cookie_insert_var() failed." );

    fh = pvo_malloc( sizeof(*fh) );
    if( -1 == pvo_file_create( "test2", PVO_FILE_BYTE_ORDER_MACHINE, cookie, fh ))
        PVO_DIE( "pvo_file_create() failed." );
    if( -1 == pvo_file_delete( fh ))
        PVO_DIE( "pvo_file_delete() failed." );
    pvo_free( fh );

    if( -1 == pvo_cookie_delete( cookie ))
        PVO_DIE( "pvo_cookie_delete() failed." );

    pvo_quit();
    MPI_Finalize();

    return 0;
}

File: test/pvo_file.f90

! vi: tabstop=4:expandtab:hlsearch

program pvo_file_f
    use mpi
    use pvo
    implicit none

    integer :: ierr, num
    integer(kind = PVO_COOKIE_KIND) :: cookie
    integer(kind = PVO_FILE_KIND) :: fh
    integer, dimension(2) :: p_i32
    real*8, dimension(2) :: p_f64

    call mpi_init( ierr )
    call pvo_init( MPI_COMM_WORLD, ierr )

    call pvo_cookie_create( mod(pvo_world_rank(),2), cookie, ierr )
    if( 0 .ne. ierr ) then
        write(*,*)  'pvo_cookie_create failed'
        call mpi_abort( MPI_COMM_WORLD, 1, ierr )
    endif

    call pvo_cookie_insert_var_int32( cookie, PVO_VAR_NODEDATA, 8, "I32"//char(0), p_i32, ierr )
    if( 0 .ne. ierr ) then
        write(*,*)  'pvo_cookie_insert_var_int32 failed'
        call mpi_abort( MPI_COMM_WORLD, 1, ierr )
    endif
    
    call pvo_cookie_insert_var_float64( cookie, PVO_VAR_CELLDATA, 8, "F64"//char(0), p_f64, ierr )
    if( 0 .ne. ierr ) then
        write(*,*)  'pvo_cookie_insert_var_float64 failed'
        call mpi_abort( MPI_COMM_WORLD, 1, ierr )
    endif

    call pvo_file_create( "test10", PVO_FILE_BYTE_ORDER_MACHINE, cookie, fh, ierr )
    if(0 .ne. ierr) then
        write(*,*) 'pvo_file_create failed'
        call mpi_abort( MPI_COMM_WORLD, 1, ierr )
    endif

    call pvo_file_delete( fh, ierr )
    if(0 .ne. ierr) then
        write(*,*) 'pvo_file_delete failed'
        call mpi_abort( MPI_COMM_WORLD, 1, ierr )
    endif

    call pvo_cookie_delete( cookie, ierr )
    if( 0 .ne. ierr ) then
        write(*,*) 'pvo_cookie_delete failed'
        call mpi_abort( MPI_COMM_WORLD, 1, ierr )
    endif
    

    call pvo_quit( ierr )
    call mpi_finalize( ierr )

end program

